source/val/BasicBlock.h

```diff
@@ -89,6 +89,7 @@ class BasicBlock {
 
   /// Sets the type of the BasicBlock
   void set_type(BlockType type) {
+    if (type == kBlockTypeUndefined) type_.reset();
     type_.set(type);
```

Comment:
This can't be right.   When calling set_type with Undefined, then you still get that first bit set.  Maybe do the type_.set(type) in the else clause.